src/ripple/basics/Buffer.h

```diff
@@ -40,8 +40,29 @@ class Buffer
 
 public:
     Buffer() = default;
-    Buffer (Buffer const&) = delete;
-    Buffer& operator= (Buffer const&) = delete;
+
+    Buffer& operator= (Buffer const& other)
+    {
+        size_ = other.size_;
+        if (size_)
+        {
+            p_.reset (new std::uint8_t[size_]);
+            memcpy (p_.get(), other.p_.get(), size_);
+        }
+        else
+            p_.reset();
+        return *this;
+    }
+
+    Buffer (Buffer const& other)
+        : size_ (other.size_)
+    {
+        if (size_)
+        {
+            p_.reset (new std::uint8_t[size_]);
```

Comment:
Actually this one is exception safe.  Though it is true `size_` will have the wrong value, it doesn't matter in this case because the `Buffer` never gets constructed.  Valid criticism on the copy assignment operator though.